src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ServiceProviderServiceExtensions.cs

```diff
@@ -18,6 +19,7 @@ public static class ServiceProviderServiceExtensions
         /// <typeparam name="T">The type of service object to get.</typeparam>
         /// <param name="provider">The <see cref="IServiceProvider"/> to retrieve the service object from.</param>
         /// <returns>A service object of type <typeparamref name="T"/> or null if there is no such service.</returns>
+        [return: MaybeNull]
```

Comment:
I realized I missed adding these annotations, but regening the ref assembly using `/t:GenerateReferenceSource=true` doesn't appear to pick these up. What am I doing wrong?

Reply:
cc: @safern

Reply:
You need to make sure to run it without incremental build turned on so that it makes sure it is running against the latest binary.

```
dotnet build --no-incremental /t:GenerateReferenceSource
```

in the src directory for this library.

Reply:
I think it's because this is a ns2.0 only library. The attribute is internal to the assembly, so doesn't appear as part of the ref assembly surface. `notnull` constraints show up, which is nice

Reply:
Hmm, probably that affects. Could you try targeting net5.0 and generating the reference source with net5.0 as the tfm and see if that preserves the attributes?

Reply:
That appears to work. Thanks!